Geometry_Engine/Query/Centroid.cs

```diff
@@ -26,16 +26,44 @@
 using System;
 using BH.Engine.Reflection;
 using BH.oM.Reflection.Attributes;
+using BH.oM.Geometry.ShapeProfiles;
+using BH.oM.Base;
 
 namespace BH.Engine.Geometry
 {
     public static partial class Query
     {
+        /***************************************************/
+        /**** Public Methods - Misc                     ****/
+        /***************************************************/
+
+        public static Point Centroid(this PlanarSurface surface, double tolerance = Tolerance.Distance)
+        {
+            return Centroid(new List<ICurve> { surface.ExternalBoundary }, surface.InternalBoundaries, tolerance);
+        }
+
+        /***************************************************/
+
+
+        public static Point Centroid(this IProfile profile, double tolerance = Tolerance.Distance)
+        {
+            return Centroid(profile.Edges, new List<ICurve>(), tolerance);
+        }
+
```

Comment:
Do not think we should place this here. Even if profiles right now live in Geometry_oM they are planned to be migrated to Spation_oM, (see https://github.com/BHoM/BHoM/issues/756).

(Also, the profile edges would need to be checked for inner/outer as well, but that is another point)

Reply:
Plus this method assumes that all the profile does not have holes - what about RHS for example?